chain/ethereum.ts

```diff
@@ -27,18 +27,23 @@ export namespace ethereum {
    *
    * Big enough to fit any pointer or native `this.data`.
    */
-  export type ValuePayload = u64
+  export type ValuePayload = usize
 
   /**
    * A dynamically typed value used when accessing Ethereum data.
    */
   export class Value {
-    kind: ValueKind
-    data: ValuePayload
+    readonly kind: ValueKind
+    readonly data: ValuePayload
+
+    protected constructor(kind: ValueKind, data: ValuePayload) {
```

Comment:
Is this constructor needed though? What problems did you face so you had to create it?

I faced the following issue on `graph-node` runtime tests:

```typescript
let token: Token;
token.kind = TokenKind.ADDRESS;
token.data = changetype<u32>(address);
return token
```

Code like the one above would break because the first line would leave a null pointer, so to fix I just used the default constructor AssemblyScript already gives:

```typescript
let token = new Token();
```

Reply:
I mean, maybe what you've introduced is just more safe and idiomatic. I'm not very experienced in AssemblyScript haha

Reply:
Yeah, it's more idiomatic since doesn't make sense to create instances of Value without kind and data.

Reply:
I understand that `Value` is a auxiliary class to generalize values manipulation, so to fix that test you should add a constructor to `Token` class, similarly to the one in `Value` class and instantiate it like:

 ```typescript
 let token = new Token(TokenKind.ADDRESS, changetype<u32>(address))
 ```

Reply:
Also create a new instance of a class without `new` operator isn't a good practice and smells bad